doc/examples/transform/plot_tps_deformation.py

```diff
@@ -0,0 +1,148 @@
+To deform the image, the displacement of every pixel is needed.
+In our image, we define 6 source and destination points labelled 1-6: 1-4 in the image
+corners, 5 near the smile corner and 6 in an eye. At the corners, there are no
+displacements. The smile corner moved upward and the eye position moved down.
+
+Thin Plate Splines provides a very handy interpolator for image deformation.
+"""
+import matplotlib.pyplot as plt
+import numpy as np
+
+import skimage as ski
+
+astronaut = ski.data.astronaut()
+
+src = np.array([[50, 50], [400, 50], [50, 400], [400, 400], [240, 150], [200, 100]])
+dst = np.array([[50, 50], [400, 50], [50, 400], [400, 400], [276, 100], [230, 100]])
+
+# Fit the thin plate spline from output to input
+
+warped_img = ski.transform.tps_warp(
+    astronaut, src[:, ::-1], dst[:, ::-1], grid_scaling=1
+)
+
+
+fig, axs = plt.subplots(1, 2)
+
+labels = [
+    '1',
+    '2',
+    '3',
+    '4',
+    '5',
+    '6',
+]  # Adjust the number of labels to match the number of points
+
+axs[0].imshow(astronaut, cmap='gray')
+axs[0].scatter(src[:, 0], src[:, 1], marker='x', color='cyan')
+
+for i, label in enumerate(labels):
+    axs[0].annotate(
+        label,
+        (src[:, 0][i], src[:, 1][i]),
+        textcoords="offset points",
+        xytext=(0, 5),
+        ha='center',
+        color='red',
+    )
+
+axs[1].imshow(warped_img, cmap='gray')
+axs[1].scatter(dst[:, 0], dst[:, 1], marker='x', color='cyan')
+
+for i, label in enumerate(labels):
+    axs[1].annotate(
+        label,
+        (dst[:, 0][i], dst[:, 1][i]),
+        textcoords="offset points",
+        xytext=(0, 5),
+        ha='center',
+        color='red',
+    )
+
+plt.show()
+
+######################################################################
+#
+# Interpolation
+# =============
+# In this example thin-plate spline is applied to each source and target
+# coordinate to derive an interpolation function and coefficients for
+# each target point. These coefficients are then used to transform an arbitrary
+# point associated with the reference to an interpolated location on the target.
+
+import matplotlib.pyplot as plt
+
+import skimage as ski
+
+samp = np.linspace(-2, 2, 4)
+xx, yy = np.meshgrid(samp, samp)
+
+# Make source points
+source_xy = np.column_stack((xx.ravel(), yy.ravel()))
+
+# Make target points
+yy[:, [0, 3]] *= 2
+target_xy = np.column_stack((xx.ravel(), yy.ravel()))
+
+
+# Compute the coefficient
+trans = ski.transform.TpsTransform()
+trans.estimate(source_xy, target_xy)
+
+
+# Make another arbitiary point
+samp2 = np.linspace(-1.8, 1.8, 10)
+test_xy = np.tile(samp2, [2, 1]).T
+
+# Estimate transformed points from given sets of source and targets points.
+transformed_xy = trans(test_xy)
+
+fig, (ax0, ax1) = plt.subplots(nrows=1, ncols=2, figsize=(8, 3))
+
+ax0.scatter(source_xy[:, 0], source_xy[:, 1], label='Source Points')
+ax0.scatter(test_xy[:, 0], test_xy[:, 1], c="orange", label='Test Points')
+ax0.legend(loc="upper center")
```

Comment:
```suggestion
ax0.scatter(test_xy[:, 0], test_xy[:, 1], c='orange', label='Test Points')
ax0.legend(loc='upper center')
```
(visual consistency)